astropy/utils/decorators.py

```diff
@@ -27,6 +29,55 @@
 ]
 
 _NotFound = object()
+_method_types = (classmethod, staticmethod, types.MethodType)
+
+
+def _deprecate_doc(old_doc, message, since):
+    """Returns a given docstring with a deprecation message prepended."""
+    old_doc = textwrap.dedent(old_doc).strip("\n") if old_doc else ""
+    new_doc = f"\n.. deprecated:: {since}\n    {message.strip()}\n\n" + old_doc
+    if not old_doc:
+        # This is to prevent a spurious 'unexpected unindent' warning from
+        # docutils when the original docstring was blank.
+        new_doc += r"\ "
+    return new_doc
+
+
+def _get_func(func):
+    """
+    Given a function or classmethod (or other function wrapper type), get
+    the function object.
+    """
+    return func.__func__ if isinstance(func, _method_types) else func
+
+
+def _deprecation_message(alternative, pending):
```

Comment:
In a followup PR this can be made a public function so people can have an Astropy-style deprecation message but use core python for their deprecations.